src/kitovu/sync/filecache.py

```diff
@@ -0,0 +1,70 @@
+"""FileCache keeps track of the state of the files, remotely and locally. It exists to determine if the local file
+has been changed between two synchronisation processes and allows for a conflict handling accordingly."""
+
+from enum import Enum
+import json
+import pathlib
+import typing
+
+from kitovu.sync import syncplugin
+from kitovu import utils
+
+# sync-process:
+# every time we sync something, the FileCache gets read, upon which kitovu decides what needs to be downloaded.
+# The files get downloaded, for each file, the FileCache gets updated for the files that were synchronised.
+
+
+class Filestate(Enum):
+    """Used to discern in which places files have changed"""
+    LOCAL = 1
+    REMOTE = 2
+    BOTH = 3
+    NONE = 4
+
+
+class File:
+    def __init__(self, local_digest_at_synctime: str,
+                 relative_path_with_filename: pathlib.PurePath,
+                 plugin: syncplugin.AbstractSyncPlugin) -> None:
+        file_properties = {"path": relative_path_with_filename,
```

Comment:
I don't understand what you're trying to do here. All this data should probably be saved in separate attributes (`self._relative_path_with_filename`, `self._local_digest_at_synctime`, etc.).

If you need some dict-structure like this for JSON, you can add a `to_dict()` method, but it doesn't make much sense to store it like that internally I think.

Reply:
Initially I had this construct as you suggested, then I changed it, since I had the idea, that it is easy to parse a dict to json. I'll revert to the original idea then.